src/ring_theory/multiplicity.lean

```diff
@@ -127,11 +127,23 @@ eq_some_iff.2 ⟨dvd_refl _, mt is_unit_iff_dvd_one.2 $ by simpa⟩
 get_eq_iff_eq_some.2 (eq_some_iff.2 ⟨dvd_refl _,
   by simpa [is_unit_iff_dvd_one.symm] using not_unit_of_finite ha⟩)
 
+lemma is_unit_right {a b : α} (ha : ¬is_unit a) (hb : is_unit b) :
+  multiplicity a b = 0 :=
+eq_some_iff.2 ⟨by simp, by { rw pow_one, exact λ h, mt (is_unit_of_dvd_unit h) ha hb, }⟩
+
 @[simp] lemma multiplicity_unit {a : α} (b : α) (ha : is_unit a) : multiplicity a b = ⊤ :=
 eq_top_iff.2 (λ _, is_unit_iff_forall_dvd.1 (ha.pow _) _)
 
 @[simp] lemma one_left (b : α) : multiplicity 1 b = ⊤ := by simp [eq_top_iff]
 
+lemma multiplicity.unit_left {a: α}
+  (u: units α): multiplicity (u: α) a = ⊤ :=
+multiplicity_unit a (is_unit_unit u)
```

Comment:
The name of this lemma is now `multiplicity.multiplicity.unit_left`, which I suspect is not what you want.
```suggestion
lemma unit_left (a : units α) (b : α) : multiplicity (a : α) b = ⊤ :=
multiplicity_unit b (is_unit_unit a)
```
I'd probably recommend renaming `multiplicity_unit` to `is_unit_left` too.

Reply:
`multiplicity_unit` isn't mine, renaming it would cause breaks in users' proofs... I don't know how mathlib handles deprecation.

Reply:
That's a sensible thing to worry about generally, but mathlib doesn't care about that at all (as long as you point out that you renamed it in the PR description)

Reply:
Ok, I go for this change then